### ISSUE TITLE
Cannot trigger 304 Response. If-Modified-Since and If-None-Match seems to be ignored

### ISSUE THREAD

**Issue by HALOCORE** [pallets/quart#172]
**Cannot trigger 304 Response. Seems not implemented.**

```py
@app.route('/static/<path:filepath>')
async def static_file(filepath):
  resp = await send_from_directory('static', filepath, conditional=True)
  resp.headers.set('Cache-Control', 'no-cache') #'public, max-age=43200,
  return resp
```

On repreating requests,  browser adds `If-Modified-Since` and `If-None-Match` in request:
```
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en,zh-CN;q=0.9,zh;q=0.8
Connection: keep-alive
Cookie: lang=en-US
Host: XXXX
If-Modified-Since: Sat, 27 Aug 2022 07:41:20 GMT
If-None-Match: "1661586080.2259881-17759-802489819"
...
```

But server still respond a 200 with same date and ETag
```
accept-ranges: bytes
cache-control: no-cache
Connection: keep-alive
Content-Length: 17759
Content-Type: application/javascript; charset=utf-8
Date: Sat, 27 Aug 2022 08:35:50 GMT
etag: "1661586080.2259881-17759-802489819"
expires: Sat, 27 Aug 2022 20:35:50 GMT
last-modified: Sat, 27 Aug 2022 07:41:20 GMT
```

**Expected behavior:** 
Return a 304 response.

**Environment:**

- Python version: 3.9
- Quart version: 0.18

May I ask is this a bug or just expected behavior? Thanks!

**Comment by pgjones** [comment]
Does it work the way you expect with Flask? (I can't remember at the moment if this is expected or not)

**Comment by HALOCORE** [comment]
Yes it works as expected in Flask. It will return 304 on following requests.
```
HTTP/1.1 304 NOT MODIFIED
Server: Werkzeug/2.2.2 Python/3.9.13
Date: Thu, 01 Sep 2022 14:03:11 GMT
Content-Disposition: inline; filename=index.js
Cache-Control: no-cache
ETag: "1662037735.3273335-24765-1464666109"
Date: Thu, 01 Sep 2022 14:03:11 GMT
Access-Control-Allow-Origin: *
Connection: close
```

**Comment by pgjones** [comment]
This should be fixed in ece2e7d779e8bcbf244c93791bc23276f955f3ca
